### PR TITLE
Remove global `-Wno-unused-parameter` warning suppression

### PR DESCRIPTION
**Pull request by freak82**
Add the suppression to `wasm/lib/CMakeLists.txt` because the source code there can't be modified but there are two unused parameters in `signature_util.cc`

Built different presets with both GCC and Clang. Built all conditionally built plugins. 
Let me try again and see if I'm missing some other build configuration.

### CODE REVIEW

File: plugins/experimental/wasm/lib/CMakeLists.txt [hunk 1]
```diff
@@ -31,6 +31,7 @@ if(wamr_FOUND)
   list(APPEND CC_FILES src/wamr/wamr.cc)
 endif()
 
+add_compile_options(-Wno-unused-parameter)
 add_library(wasmlib STATIC ${CC_FILES})
 if(wamr_FOUND)
   target_compile_options(wasmlib PRIVATE -Wno-missing-field-initializers)
```
Comment (JosiahWI) [inline plugins/experimental/wasm/lib/CMakeLists.txt:37]:
```suggestion
add_library(wasmlib STATIC ${CC_FILES})
target_compile_options(wasmlib PUBLIC -Wno-unused-parameter)
if(wamr_FOUND)
  target_compile_options(wasmlib PRIVATE -Wno-missing-field-initializers)
```

**Review by JosiahWI — CHANGES_REQUESTED** [review]
Could we do it this way? We should be careful to limit flags in scope when possible; we don't want to unintentionally suppress the warning in places we don't have to.

File: plugins/experimental/wasm/lib/CMakeLists.txt (hunk 1 above)
Comment (freak82) [reply plugins/experimental/wasm/lib/CMakeLists.txt]: Agreed.

**Review by JosiahWI — COMMENTED** [review]
Looks good, will re-examine for approval once we restore CI.

**Comment by JosiahWI** [comment]
[approve ci]

**Comment by JosiahWI** [comment]
[approve ci]

**Comment by ezelkow1** [comment]
[approve ci]

**Comment by freak82** [comment]
Can't see the reason for the above failures in the console outputs. I'm not sure why.
I see only information like this
```
Build [Github_Builds » rocky #5437](https://ci.trafficserver.apache.org/job/Github_Builds/job/rocky/5437/) completed: FAILURE
[Pipeline] echo
Build of 'Github_Builds/rocky' returned result: FAILURE
[Pipeline] setGitHubPullRequestStatus
Setting pull request status Rocky to FAILURE with message: Failure - https://ci.trafficserver.apache.org/job/Github_Builds/job/rocky/5437/
```
Am I missing something again?

**Comment by JosiahWI** [comment]
On Rocky:
```
../src/iocore/net/quic/QUICStreamManager.cc:105:79: error: unused parameter 'err' [-Werror=unused-parameter]
  105 | QUICStreamManager::create_stream(QUICStreamId stream_id, QUICConnectionError &err)
  ```
  
  On OSX:
  ```
  /Users/jenkins/workspace/Github_Builds/osx/src/plugins/experimental/fq_pacing/fq_pacing.cc:200:22: error: unused parameter 'instance' [-Werror,-Wunused-parameter]
TSRemapDoRemap(void *instance, TSHttpTxn txnp, TSRemapRequestInfo * /* rri ATS_UNUSED */)
```
  
  On Fedora:
  ```
  ../example/cripts/example1.cc: In function 'void _do_init(TSRemapInterface*)':
../include/cripts/Preamble.hpp:49:62: error: unused parameter 'api_info' [-Werror=unused-parameter]
   49 | #define do_init()            void _do_init(TSRemapInterface *api_info)
      |                                            ~~~~~~~~~~~~~~~~~~^~~~~~~~
../example/cripts/example1.cc:29:1: note: in expansion of macro 'do_init'
   29 | do_init()
      | ^~~~~~~
In file included from ../example/cripts/example1.cc:255:
../include/cripts/Epilogue.hpp: In instantiation of 'bool wrap_post_remap(T*, bool, CaseTag<0>) [with T = Cript::Context]':
../include/cripts/Epilogue.hpp:342:20:   required from here
../include/cripts/Preamble.hpp:49:62: error:   342 |     wrap_post_remap(context, true, CaseArg);
../include/cripts/Preamble.hpp:49:62: error:       |     ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
../include/cripts/Epilogue.hpp:84:20: error: unused parameter 'context' [-Werror=unused-parameter]
   84 | wrap_post_remap(T *context, bool execute, CaseTag<0>) -> bool
      |                 ~~~^~~~~~~
../include/cripts/Epilogue.hpp:84:34: error: unused parameter 'execute' [-Werror=unused-parameter]
   84 | wrap_post_remap(T *context, bool execute, CaseTag<0>) -> bool
      |                             ~~~~~^~~~~~~
../include/cripts/Epilogue.hpp: In instantiation of 'bool wrap_delete_instance(T*, bool, CaseTag<0>) [with T = Cript::InstanceContext]':
../include/cripts/Epilogue.hpp:448:70:   required from here
../include/cripts/Epilogue.hpp:84:34: error:   448 |   bool                   needs_delete_instance = wrap_delete_instance(&context, false, CaseArg);
../include/cripts/Epilogue.hpp:84:34: error:       |                                                  ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
../include/cripts/Epilogue.hpp:231:25: error: unused parameter 'context' [-Werror=unused-parameter]
  231 | wrap_delete_instance(T *context, bool execute, CaseTag<0>) -> bool
      |                      ~~~^~~~~~~
../include/cripts/Epilogue.hpp:231:39: error: unused parameter 'execute' [-Werror=unused-parameter]
  231 | wrap_delete_instance(T *context, bool execute, CaseTag<0>) -> bool
  ```
  
  On CentOS:
  ```
  ../../../../src/iocore/net/UnixUDPNet.cc:87:43: error: unused parameter 'send_at_hint' [-Werror=unused-parameter]
   87 |                          struct timespec *send_at_hint)
      |                          ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
```

**Comment by freak82** [comment]
Thank you.
I'll close this pull request.
Don't know how I missed these especially rocky and fedora presets which I built.

**Comment by JosiahWI** [comment]
[approve ci]

**Comment by JosiahWI** [comment]
On OSX:

```
/Users/jenkins/workspace/Github_Builds/osx/src/src/tscore/hugepages.cc:64:23: error: unused parameter 'enabled' [-Werror,-Wunused-parameter]
ats_hugepage_init(int enabled)
```

On AuTest:

```
../tests/tools/plugins/conf_remap_stripped.cc: In function 'TSReturnCode TSRemapInit(TSRemapInterface*, char*, int)':
../tests/tools/plugins/conf_remap_stripped.cc:26:31: error: unused parameter 'api_info' [-Werror=unused-parameter]
   26 | TSRemapInit(TSRemapInterface *api_info, char *errbuf, int errbuf_size)
      |             ~~~~~~~~~~~~~~~~~~^~~~~~~~
../tests/tools/plugins/conf_remap_stripped.cc:26:47: error: unused parameter 'errbuf' [-Werror=unused-parameter]
   26 | TSRemapInit(TSRemapInterface *api_info, char *errbuf, int errbuf_size)
      |                                         ~~~~~~^~~~~~
../tests/tools/plugins/conf_remap_stripped.cc:26:59: error: unused parameter 'errbuf_size' [-Werror=unused-parameter]
   26 | TSRemapInit(TSRemapInterface *api_info, char *errbuf, int errbuf_size)
      |                                                       ~~~~^~~~~~~~~~~
../tests/tools/plugins/conf_remap_stripped.cc: In function 'TSReturnCode TSRemapNewInstance(int, char**, void**, char*, int)':
../tests/tools/plugins/conf_remap_stripped.cc:32:24: error: unused parameter 'argc' [-Werror=unused-parameter]
   32 | TSRemapNewInstance(int argc, char *argv[], void **ih, char * /* errbuf ATS_UNUSED */, int /* errbuf_size ATS_UNUSED */)
      |                    ~~~~^~~~
../tests/tools/plugins/conf_remap_stripped.cc:32:36: error: unused parameter 'argv' [-Werror=unused-parameter]
   32 | TSRemapNewInstance(int argc, char *argv[], void **ih, char * /* errbuf ATS_UNUSED */, int /* errbuf_size ATS_UNUSED */)
      |                              ~~~~~~^~~~~~
../tests/tools/plugins/conf_remap_stripped.cc:32:51: error: unused parameter 'ih' [-Werror=unused-parameter]
   32 | TSRemapNewInstance(int argc, char *argv[], void **ih, char * /* errbuf ATS_UNUSED */, int /* errbuf_size ATS_UNUSED */)
      |                                            ~~~~~~~^~
../tests/tools/plugins/conf_remap_stripped.cc: In function 'void TSRemapDeleteInstance(void*)':
../tests/tools/plugins/conf_remap_stripped.cc:38:29: error: unused parameter 'ih' [-Werror=unused-parameter]
   38 | TSRemapDeleteInstance(void *ih)
      |                       ~~~~~~^~
../tests/tools/plugins/conf_remap_stripped.cc: In function 'TSRemapStatus TSRemapDoRemap(void*, TSHttpTxn, TSRemapRequestInfo*)':
../tests/tools/plugins/conf_remap_stripped.cc:46:22: error: unused parameter 'ih' [-Werror=unused-parameter]
   46 | TSRemapDoRemap(void *ih, TSHttpTxn rh, TSRemapRequestInfo * /* rri ATS_UNUSED */)
      |                ~~~~~~^~
../tests/tools/plugins/conf_remap_stripped.cc:46:36: error: unused parameter 'rh' [-Werror=unused-parameter]
   46 | TSRemapDoRemap(void *ih, TSHttpTxn rh, TSRemapRequestInfo * /* rri ATS_UNUSED */)
```

@freak82 It looks like most of the CI builds are passing now. OSX and AuTest are still unhappy. :+1:

**Comment by freak82** [comment]
Thank you for the help.
I'll check these.

I built OSX like
```
cmake --preset ci-osx -DCMAKE_C_COMPILER=/usr/bin/clang-17 -DCMAKE_CXX_COMPILER=/usr/bin/clang++-17 -DENABLE_MAGICK=ON -DENABLE_MAXMIND_ACL=ON -DENABLE_OTEL_TRACER=ON -DENABLE_STEK_SHARE=ON -DENABLE_URI_SIGNING=ON -DENABLE_WASM=ON

cmake --build ./build-ci 2>&1 | tee /z/trafficserver/build-ci/build.log
```
but obviously I'm still missing some defines and build flags/configurations here and there.

**Comment by JosiahWI** [comment]
Exhaustively testing build configurations is hard. It could come down to the existence of support for various features in the OS, to name one possibility. We seem to be very close now, and I don't know of a good way to catch these last cases locally without the right system or special setup, so I think it's fine to let CI catch the last few for us.

**Comment by JosiahWI** [comment]
[approve ci]

**Comment by JosiahWI** [comment]
On OSX:

```
/Users/jenkins/workspace/Github_Builds/osx/src/src/tscore/ink_cap.cc:89:43: error: unused parameter 'suid' [-Werror,-Wunused-parameter]
getresuid(uid_t *uid, uid_t *euid, uid_t *suid)
                                          ^
/Users/jenkins/workspace/Github_Builds/osx/src/src/tscore/ink_cap.cc:99:43: error: unused parameter 'sgid' [-Werror,-Wunused-parameter]
getresgid(gid_t *gid, gid_t *egid, gid_t *sgid)
                                          ^
/Users/jenkins/workspace/Github_Builds/osx/src/src/tscore/ink_cap.cc:336:21: error: unused parameter 'flag' [-Werror,-Wunused-parameter]
EnableCoreFile(bool flag)
```

Perhaps it needs `-DENABLE_POSIX_CAP`?

The ParentSelection regression test failed on CentOS. Although concerning, this is a transient failure and unrelated to these changes.

**Comment by freak82** [comment]
Ok. Thank you. I'll try with `-DENABLE_POSIX_CAP` to see if the build errors are reproducible.

**Comment by JosiahWI** [comment]
[approve ci]

**Comment by JosiahWI** [comment]
On OSX:
```
/Users/jenkins/workspace/Github_Builds/osx/src/src/iocore/net/UnixNetVConnection.cc:1514:52: error: unused parameter 'side' [-Werror,-Wunused-parameter]
UnixNetVConnection::set_tcp_congestion_control(int side)
```

**Comment by freak82** [comment]
Thank you.
This is related to a parameter which is unused in case `TCP_CONGESTION` is not defined.
As far as I checked this define is used in two places and the other one is correct.

Should I do a separate pull request for this single change or I should do apply it here?

**Comment by JosiahWI** [comment]
There are more OSX specific issues than I expected. I have access to an OSX machine, so I will check out your branch and see how many warnings are left.

**Comment by freak82** [comment]
If you can post somewhere all of the warnings I can fix them and do a separate pull request for them then.

**Comment by JosiahWI** [comment]
After adding `[[maybe_unused]]` to the parameter for when `TCP_CONGESTION` is undefined, it built successfully. I did have to disable some experimental plugins for things I don't have installed, but those are high level so they shouldn't have much platform specific code. Could you make a PR for that one change?

**Comment by JosiahWI** [comment]
[approve ci osx]

**Comment by freak82** [comment]
🎉

**Review by JosiahWI — APPROVED** [review]
Thank you for working on this. The warning can help catch mistakes.
